src/www/app/server/access_config.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright 2018 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {makeConfig, SHADOWSOCKS_URI, SIP002_URI} from 'ShadowsocksConfig';
+import {ServerUrlInvalid, ServerIncompatible, ShadowsocksUnsupportedCipher} from '../../model/errors';
+import {ShadowsocksConfig} from '../config';
+
+export class OutlineServerAccessConfig implements ShadowsocksConfig {
+  // We restrict to AEAD ciphers because unsafe ciphers are not supported in go-tun2socks.
+  // https://shadowsocks.org/en/spec/AEAD-Ciphers.html
+  private static readonly SUPPORTED_CIPHERS = new Set([
+    'chacha20-ietf-poly1305',
+    'aes-128-gcm',
+    'aes-192-gcm',
+    'aes-256-gcm',
+  ]);
+
+  host?: string;
```

Comment:
Shall we make these properties `readonly`?

Reply:
I'm not sure yet 😅 I think `private set` works because we'll need to update them on a config location sync

Reply:
Updated!

Reply:
Configs are supposed to be immutable Value Objects.